ql/src/java/org/apache/hadoop/hive/ql/io/parquet/vector/VectorizedListColumnReader.java

```diff
@@ -129,21 +131,16 @@ private boolean fetchNextValue(PrimitiveObjectInspector.PrimitiveCategory catego
   private void addElement(ListColumnVector lcv, List<Object> elements, PrimitiveObjectInspector.PrimitiveCategory category, int index) throws IOException {
     lcv.offsets[index] = elements.size();
 
-    // Return directly if last value is null
-    if (definitionLevel < maxDefLevel) {
-      lcv.isNull[index] = true;
-      lcv.lengths[index] = 0;
-      // fetch the data from parquet data page for next call
-      fetchNextValue(category);
-      return;
-    }
-
     do {
       // add all data for an element in ListColumnVector, get out the loop if there is no data or the data is for new element
+      if (definitionLevel < maxDefLevel) {
```

Comment:
the original intention here was to sign if there is a NULL value instead of a list, which is happens in definitionLevel == 0, I'll change this part and add some comments